Project/Gem/Source/Vision/IdealVisionSystem.cpp

```diff
@@ -0,0 +1,305 @@
+        AzPhysics::SceneHandle defaultSceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        const AZ::Transform sensorTransformInv = sensorTransform.GetInverse();
+
+        geometry_msgs::msg::PoseArray poseArray;
+        vision_msgs::msg::Detection2DArray detection2DArray;
+        vision_msgs::msg::Detection3DArray detection3DArray;
+
+        std_msgs::msg::Header header;
+        header.frame_id = GetFrameID().c_str();
+        header.stamp = ROS2Interface::Get()->GetROSTimestamp();
+
+        poseArray.header = header;
+        detection3DArray.header = header;
+
+        // Overlap query with frustum
+        if (m_cookedMesh && sceneInterface && defaultSceneHandle != AzPhysics::InvalidSceneHandle)
+        {
+            AzPhysics::OverlapRequest request;
+            request.m_shapeConfiguration = m_cookedMesh;
+            request.m_pose = sensorTransform;
+            request.m_filterCallback = nullptr;
+            AzPhysics::SceneQueryHits results = sceneInterface->QueryScene(defaultSceneHandle, &request);
+            AZ_Printf("IdealVisionSystem", "Overlap with %d objects", results.m_hits.size());
+
+            for (const auto& result : results.m_hits)
+            {
+                if (!m_excludeEntities.contains(result.m_entityId))
+                {
+                    const AZ::Aabb aabb = result.m_shape->GetAabbLocal();
+
+                    AZ::Transform resutlTransform;
+                    AZ::TransformBus::EventResult(resutlTransform, result.m_entityId, &AZ::TransformBus::Events::GetWorldTM);
+                    const AZ::Transform resultTransformLocal = sensorTransformInv * resutlTransform;
+
+                    AZStd::string targetName;
+                    AZ::ComponentApplicationBus::BroadcastResult(
+                        targetName, &AZ::ComponentApplicationBus::Events::GetEntityName, result.m_entityId);
+
+                    geometry_msgs::msg::Pose pose;
+                    pose.orientation = ROS2::ROS2Conversions::ToROS2Quaternion(resultTransformLocal.GetRotation());
+                    pose.position = ROS2::ROS2Conversions::ToROS2Point(resultTransformLocal.GetTranslation());
+                    poseArray.poses.push_back(pose);
+
+                    // construct 3D detection
+                    vision_msgs::msg::Detection3D detection3D;
+                    detection3D.header = header;
+                    detection3D.id = targetName.c_str();
+
+                    detection3D.bbox.center.position = pose.position;
+                    detection3D.bbox.size.x = aabb.GetXExtent() * 2.0f;
+                    detection3D.bbox.size.y = aabb.GetYExtent() * 2.0f;
+                    detection3D.bbox.size.z = aabb.GetZExtent() * 2.0f;
+
+                    detection3D.results.resize(1);
+                    detection3D.results.front().pose.pose = pose;
+                    detection3D.results.front().hypothesis.class_id = -1;
```

Comment:
Developers could use this field and it could be filled from tags.

Reply:
It will be crc32 of tag name.